src/Button/GeoLocationButton/GeoLocationButton.jsx

```diff
@@ -114,18 +153,45 @@ class GeoLocationButton extends React.Component {
    */
   constructor(props) {
     super(props);
+    const {
+      map,
+      showMarker
+    } = this.props;
 
     this.positions = new OlGeomLineString([], 'XYZM');
+    this._geoLocationLayer.setStyle(this._styleFunction);
+    map.addLayer(this._geoLocationLayer);
```

Comment:
Please check if the layer is already added to the map. In addition I would recommend to delete the layer on unmount.